client/my-sites/backup/status/hooks.js

```diff
@@ -138,10 +144,15 @@ export const useRealtimeBackupStatus = ( siteId, selectedDate ) => {
 		after: moment( selectedDate ).startOf( 'day' ).toISOString(),
 	} );
 
-	const backupAttemptsOnDate = activityLogs.filter(
-		( activity ) => isActivityBackup( activity ) || isSuccessfulRealtimeBackup( activity )
+	const backupAttemptsOnDate = useMemo(
+		() =>
+			activityLogs.filter(
+				( activity ) => isActivityBackup( activity ) || isSuccessfulRealtimeBackup( activity )
+			),
+		[ activityLogs ]
 	);
 	const lastBackupAttemptOnDate = backupAttemptsOnDate[ 0 ];
+	const lastSuccessfulBackupOnDate = backupAttemptsOnDate.filter( isSuccessfulRealtimeBackup )[ 0 ];
```

Comment:
Could it be possible to group these three vars together into a single memoized loop? I'm worried here as I know AL are prettyyyy big arrays.

Reply:
My impression was that these calculations would only happen whenever `backupAttemptsOnDate` changes, but my understanding of how React handles re-renders is limited. Would the two non-memoized vars always be re-calc'ed even if the array they're derived from is the same?

Reply:
I believe so yes, as when the component unmounts these vars are garbage collected, and would need to be recalculated on remount.

Reply:
I put several vars into a memoized function call in my latest commit :-)